pytorch_lightning/overrides/data_parallel.py

```diff
@@ -160,7 +160,17 @@ def parallel_apply(self, replicas, inputs, kwargs):
         return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])
 
     def forward(self, *inputs, **kwargs):  # pragma: no-cover
-        self._sync_params()
+
+        # --------------
+        # LIGHTNING MOD
+        # --------------
+        # normal
+        # self._sync_params()
+        #
+        # not necessary, when parameters did not change. 
+        # this is true, if the last step was doing backward
+        if self.module.trainer.batch_idx % self.module.trainer.accumulate_grad_batches == 0:
```

Comment:
```suggestion
        if (self.module.trainer.batch_idx + 1) % self.module.trainer.accumulate_grad_batches == 0:
```
since batch_idx is 0 indexed, and we do it this way also in the other places for accumulation